src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -312,28 +312,22 @@ public static ZipArchive ReadZipFile (string filename, bool strictConsistencyChe
 				}
 				var outfile = Path.GetFullPath (Path.Combine (destination, fullName));
 				files.Add (outfile);
-				var dt = File.Exists (outfile) ? File.GetLastWriteTimeUtc (outfile) : DateTime.MinValue;
```

Comment:
We were doing a `File.Exists` and reading the timestamp of every output file. That was slower than computing a hash of each assembly, since each assembly contains a zip with lots of files.